internal/model/date.go

```diff
@@ -0,0 +1,82 @@
+package model
+
+import (
+	"strings"
+	"time"
+)
+
+type Date struct {
+	Time time.Time
+}
+
+func NewDate(d string) Date {
+	t, err := stringToTime(d)
+	if err != nil {
+		panic(err)
```

Comment:
this should return the error, you want to avoid using `panic` unless you are really sure this is never going to get bad data

Reply:
We are sure that this will not get bad data